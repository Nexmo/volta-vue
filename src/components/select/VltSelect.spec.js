import { shallowMount } from '@vue/test-utils';
import { VltSelect } from '..';

describe('vlt-select', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(VltSelect, {
      propsData: {
        val: undefined,
        options: ['red', 'yellow', 'blue'],
      },
    });
  });

  test('renders correctly with options "red", "yellow", "blue"', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('selected value is "blue"', () => {
    wrapper
      .findAll('option')
      .at(2)
      .setSelected();
    expect(wrapper.emitted('input')).toEqual([['blue']]);
  });
});
