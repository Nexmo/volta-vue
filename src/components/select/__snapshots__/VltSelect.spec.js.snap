// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-select renders correctly with options "red", "yellow", "blue" 1`] = `
<div
  class="Vlt-select"
>
  <select>
    <!--v-if-->
    
    <option
      value="red"
    >
      red
    </option>
    <option
      value="yellow"
    >
      yellow
    </option>
    <option
      value="blue"
    >
      blue
    </option>
    
  </select>
  <!--v-if-->
</div>
`;
